source/Ldap.Integration.Tests/scripts/ActiveDirectory/Azure/New-ActiveDirectoryIntegrationTestEnvironment.ps1

```diff
@@ -0,0 +1,16 @@
+Write-Output "Setting up Azure AD Terraform stack"
+
+terraform init
+terraform plan
+terraform apply --auto-approve
+
+#We ned to refresh the state since the public vm IP isn't allocated when the resource is first provisioned
+terraform refresh
```

Comment:
That's a surprise, skimming your main.tf it looks like the output should contain the public IP as soon as apply is done.

Reply:
I know right. Apparently it should be set but there seems to be an issue that sounds like its pretty common with Dynamic IPs in Azure. I would have thought but setting the depends in the output that it would get it once the Vm becomes available but apparently not